apps/service_define/lib/define/event/store.ex

```diff
@@ -0,0 +1,85 @@
+defmodule Define.Event.Store do
+  require Logger
+  alias Define.DefinitionSerialization
+  alias Define.Model.{DataDefinitionView, ExtractView, LoadView, TransformView}
+  import Definition, only: [identifier: 1]
+
+  @instance Define.Application.instance()
+  @collection "definitions"
+
+  def update_definition(%Transform{} = data) do
+    update_definition_field(data, :transform, &to_transform_view/1)
+  end
+
+  def update_definition(%Load{} = data) do
+    update_definition_field(data, :load, &to_load_view/1)
+  end
+
+  def update_definition(%Extract{} = data) do
+    update_definition_field(data, :extract, &to_extract_view/1)
+  end
+
+  def update_definition(data) do
+    Logger.error(fn -> "Got unexpected data definition update: #{inspect(data)}" end)
+  end
+
+  defp update_definition_field(data, key, to_view_converter) do
+    get_or_create(data)
+    |> Map.put(key, to_view_converter.(data))
+    |> persist()
+  rescue
```

Comment:
Not a big fan of using rescue like this. Can we refactor to use `:ok/:error` tuples throughout?

Reply:
I originally took us down that path but then Nick pointed out that we were adding a lot of extra work/verbosity by doing so which I found myself agree with. I forget all the particulars about why it got so gross at this point though.

Reply:
So we're going to use exceptions for control flow because of verbosity?